accounts-db/src/accounts_db.rs

```diff
@@ -2012,6 +2023,7 @@ impl AccountsDb {
             storage_access,
             scan_filter_for_shrinking,
             is_experimental_accumulator_hash_enabled: enable_experimental_accumulator_hash,
+            thread_pool_hash: make_hash_thread_pool(num_threads_for_hash_thread_pool),
```

Comment:
This will end up creating the thread pool twice right: once here in`new_with_config()` and once in the call to `default_with_account_index()`? Granted this is a startup cost only and the extra pool should go away, but still heavier than our "usual" stuff that we copy/clone at startup.

As I was working on my PR, I started refactoring things to avoid this. Those cleanup PR's are streaming in, we already shipped one and I have you on review for the next one (https://github.com/anza-xyz/agave/pull/3270) if you want to do the cleanup prior to pushing this one